src/rabbit_stomp_processor.erl

```diff
@@ -589,45 +589,59 @@ do_subscribe(Destination, DestHdr, Frame,
                 rabbit_stomp_util:consumer_tag(Frame),
             case Prefetch of
                 undefined -> ok;
-                _         -> amqp_channel:call(
-                               Channel, #'basic.qos'{prefetch_count = Prefetch})
+                _         -> amqp_channel:call(Channel,
+                                               #'basic.qos'{
+                                                  prefetch_count = Prefetch
+                                                 })
             end,
-            ExchangeAndKey = rabbit_routing_util:parse_routing(Destination),
-            try
-                amqp_channel:subscribe(Channel,
-                                       #'basic.consume'{
-                                          queue        = Queue,
-                                          consumer_tag = ConsumerTag,
-                                          no_local     = false,
-                                          no_ack       = (AckMode == auto),
-                                          exclusive    = false,
-                                          arguments    = []},
-                                       self()),
-                ok = rabbit_routing_util:ensure_binding(
-                       Queue, ExchangeAndKey, Channel)
-            catch exit:Err ->
-                    %% it's safe to delete this queue, it was server-named
-                    %% and declared by us
-                    case Destination of
-                        {exchange, _} ->
-                            ok = maybe_clean_up_queue(Queue, State);
-                        {topic, _} ->
-                            ok = maybe_clean_up_queue(Queue, State);
-                        _ ->
-                            ok
+            case dict:find(ConsumerTag, Subs) of
+                {ok, #subscription{ack_mode = AckMode}} ->
+                    Message = "Duplicated ConsumerTag",
+                    Detail = "attempt to reuse consumer tag '~s'.",
+                    log_error(Message, rabbit_misc:format(Detail, [ConsumerTag]), none),
+                    send_error(Message,
```

Comment:
This (or rather, `error/3` or `error/4`, which is reinvented here) will not terminate the connection. So an explicit exit is necessary. However, we should be careful about how we do this: we need to make sure the reader has a chance to send the error message before its supervisor terminates it (because the processor has exited, and it is an intrinsic child).